drivers/lora/sx12xx_common.c

```diff
@@ -228,6 +267,12 @@ int sx12xx_lora_send(const struct device *dev, uint8_t *data,
 	 * modem and driver.
 	 */
 	ret = k_poll(&evt, 1, K_MSEC(2 * air_time));
+	if (done.result < 0) {
```

Comment:
`done.result` is not guaranteed to be in any particular state if `k_poll` times out.
You should be using `k_poll_signal_check`